llrt_core/src/modules/http/url_search_params.rs

```diff
@@ -1,217 +1,386 @@
-                }
+    //
+    // Properties
+    //
 
-                let keys = obj.keys::<String>();
-                let key_len = keys.len();
+    #[qjs(get)]
+    pub fn size(&self) -> usize {
+        self.url.borrow().query_pairs().count()
+    }
 
-                let mut params = Vec::with_capacity(key_len);
+    //
+    // Instance methods
+    //
 
-                for key in keys {
-                    let key = key?;
-                    let val = obj.get::<_, Coerced<String>>(&key)?;
-                    params.push((key, val.to_string()))
+    pub fn append(&mut self, key: Coerced<String>, value: Coerced<String>) {
+        self.url
+            .borrow_mut()
+            .query_pairs_mut()
+            .append_pair(key.as_str(), value.as_str());
+    }
+
+    pub fn delete(&mut self, ctx: Ctx<'js>, key: Coerced<String>, value: Opt<Value<'js>>) {
+        let key = key.0;
+        let value = value.into_inner();
+
+        let new_pairs: Vec<_> = self
+            .url
+            .borrow()
+            .query_pairs()
+            .into_owned()
+            .filter(|(k, v)| {
+                if let Some(value) = value.clone() {
+                    if !value.is_undefined() {
+                        let value: Result<Coerced<String>> = Coerced::from_js(&ctx, value);
+                        if let Ok(value) = value {
+                            return !(*k == key && *v == *value);
+                        }
+                    }
                 }
-                return Ok(Self { params });
-            }
-        }
+                *k != key
+            })
+            .collect();
 
-        Ok(URLSearchParams { params: Vec::new() })
+        if !new_pairs.is_empty() {
+            self.url
+                .borrow_mut()
+                .query_pairs_mut()
+                .clear()
+                .extend_pairs(new_pairs);
+        } else {
+            self.url.borrow_mut().set_query(None);
+        }
     }
 
-    #[qjs(get)]
-    pub fn size(&self) -> usize {
-        self.params.len()
+    pub fn entries(&self, ctx: Ctx<'js>) -> Result<Value<'js>> {
+        self.js_iterator(ctx)
     }
 
-    pub fn append(&mut self, key: String, value: String) {
-        self.params.push((key, value));
+    pub fn for_each(&self, callback: Function<'js>) -> Result<()> {
+        self.url
+            .borrow()
+            .query_pairs()
+            .into_owned()
+            .try_for_each(|(k, v)| callback.call((v, k)))?;
+        Ok(())
     }
 
-    pub fn get<'js>(&mut self, ctx: Ctx<'js>, key: String) -> Result<Value<'js>> {
-        self.params
-            .iter()
-            .find(|(k, _)| k == &key)
-            .map(|(_, v)| v.into_js(&ctx))
-            .unwrap_or_else(|| Null.into_js(&ctx))
+    pub fn get(&mut self, ctx: Ctx<'js>, key: String) -> Result<Value<'js>> {
+        match self
+            .url
+            .borrow()
+            .query_pairs()
+            .into_owned()
```

Comment:
This will copy all pairs. Copy only found value. There is also a `find_map` that is usefull

Reply:
I was able to just remove `into_owned`.